From fbb68c7fd1f890f52e54a3767e3c12c070ee5a02 Mon Sep 17 00:00:00 2001
Message-Id: <fbb68c7fd1f890f52e54a3767e3c12c070ee5a02.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Mon, 29 Sep 2014 14:34:46 +0530
Subject: [PATCH 41/75] MUST_REBASE [VPG]: adf/intel/vlv: Move Sideband
 wrappers as inline in vlv_dc_config.h

Will be reused in multiple places in the modeset sequence

v2: implement wrapper for GPS_CORE and GPIO rw

v3: Move cck wrappers as inline in vlv_dc_config.h as well

v4: Correct the sideband wrappers. Most of the sideband routines need to use
    PCI_FEVFN rather than DPIO_DEVFN. Correct the same. Was working untill we
    hit S0ix patch where we need to correctly reprogram sideband registers.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I113a1c212fdee71f3e1a37cdd54c75561a1eae5e
Signed-off-by: Deepak S <deepak.s@intel.com>
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
---
 .../video/adf/intel/core/common/dsi/dsi_pipe_ops.c |    6 ---
 .../video/adf/intel/core/common/dsi/gen_dsi_pipe.c |    1 +
 .../adf/intel/core/common/dsi/intel_dsi_pll.c      |   15 +-----
 .../core/common/dsi/panels/intel_dsi_panel_vbt.c   |    6 +--
 .../adf/intel/include/core/vlv/vlv_dc_config.h     |   48 ++++++++++++++++++++
 5 files changed, 51 insertions(+), 25 deletions(-)

diff --git a/drivers/video/adf/intel/core/common/dsi/dsi_pipe_ops.c b/drivers/video/adf/intel/core/common/dsi/dsi_pipe_ops.c
index bca0f5e..5db9ea0 100644
--- a/drivers/video/adf/intel/core/common/dsi/dsi_pipe_ops.c
+++ b/drivers/video/adf/intel/core/common/dsi/dsi_pipe_ops.c
@@ -32,12 +32,6 @@
 #include "intel_dsi.h"
 #include "intel_dsi_cmd.h"
 
-static void vlv_flisdsi_write(u32 reg, u32 val)
-{
-	intel_dpio_sideband_rw(INTEL_SIDEBAND_REG_WRITE, IOSF_PORT_FLISDSI,
-			      reg, &val);
-}
-
 static void band_gap_reset(void)
 {
 	pr_err("ADF: %s\n", __func__);
diff --git a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
index 1fd335f..c1be910 100644
--- a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
+++ b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
@@ -14,6 +14,7 @@
 #include <core/vlv/vlv_dc_config.h>
 #include <core/common/dsi/dsi_pipe.h>
 #include <core/common/dsi/dsi_config.h>
+#include <core/vlv/vlv_dc_config.h>
 #include <intel_adf_device.h>
 #include "dsi_vbt.h"
 #include "intel_dsi.h"
diff --git a/drivers/video/adf/intel/core/common/dsi/intel_dsi_pll.c b/drivers/video/adf/intel/core/common/dsi/intel_dsi_pll.c
index d8ac897..b0d0da7 100644
--- a/drivers/video/adf/intel/core/common/dsi/intel_dsi_pll.c
+++ b/drivers/video/adf/intel/core/common/dsi/intel_dsi_pll.c
@@ -33,6 +33,7 @@
 #include <intel_adf_device.h>
 #include <core/common/dsi/dsi_pipe.h>
 #include <core/common/dsi/dsi_config.h>
+#include <core/vlv/vlv_dc_config.h>
 #include <core/vlv/vlv_dc_regs.h>
 #include "intel_dsi.h"
 
@@ -56,20 +57,6 @@ static const u32 lfsr_converts[] = {
 	71, 35							/* 91 - 92 */
 };
 
-static u32 vlv_cck_read(u32 reg)
-{
-	u32 val;
-	intel_dpio_sideband_rw(INTEL_SIDEBAND_REG_READ, IOSF_PORT_CCK,
-			       reg, &val);
-	return val;
-}
-
-static void vlv_cck_write(u32 reg, u32 val)
-{
-	intel_dpio_sideband_rw(INTEL_SIDEBAND_REG_WRITE, IOSF_PORT_CCK,
-			       reg, &val);
-}
-
 /* Get DSI clock from pixel clock */
 static u32 dsi_clk_from_pclk(u32 pclk, int pixel_format, int lane_count)
 {
diff --git a/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c b/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
index b3ac31f..9d75735 100644
--- a/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
+++ b/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
@@ -34,6 +34,7 @@
 #include <core/common/dsi/dsi_panel.h>
 #include <core/vlv/vlv_dc_regs.h>
 #include <core/vlv/vlv_dc_gpio.h>
+#include <core/vlv/vlv_dc_config.h>
 #include <linux/i2c.h>
 #include <linux/mfd/intel_soc_pmic.h>
 #include "intel_dsi.h"
@@ -51,11 +52,6 @@
 
 #define NS_KHZ_RATIO 1000000
 
-static void vlv_gpio_write(u32 reg, u32 val, u32 port)
-{
-	intel_dpio_sideband_rw(INTEL_SIDEBAND_REG_WRITE, port, reg, &val);
-}
-
 struct gpio_table {
 	u16 function_reg;
 	u16 pad_reg;
diff --git a/drivers/video/adf/intel/include/core/vlv/vlv_dc_config.h b/drivers/video/adf/intel/include/core/vlv/vlv_dc_config.h
index 98f37ca..8c085b6 100644
--- a/drivers/video/adf/intel/include/core/vlv/vlv_dc_config.h
+++ b/drivers/video/adf/intel/include/core/vlv/vlv_dc_config.h
@@ -16,6 +16,9 @@
 #define VLV_DC_CONFIG_H
 
 #include "core/intel_dc_config.h"
+#include "core/vlv/vlv_dc_regs.h"
+#include <drm/i915_drm.h>
+#include <drm/i915_adf.h>
 
 #define VLV_N_PLANES	6
 #define VLV_N_PIPES	2
@@ -45,6 +48,51 @@ enum vlv_disp_plane {
 	VLV_MAX_PLANES,
 };
 
+static inline void vlv_gpio_write(u32 reg, u32 val, u32 port)
+{
+	intel_adf_pci_sideband_rw(INTEL_SIDEBAND_REG_WRITE, port, reg, &val);
+}
+
+static inline void vlv_flisdsi_write(u32 reg, u32 val)
+{
+	intel_adf_dpio_sideband_rw(INTEL_SIDEBAND_REG_WRITE, IOSF_PORT_FLISDSI,
+			      reg, &val);
+}
+
+static inline void vlv_gpio_nc_write(u32 reg, u32 val)
+{
+	intel_adf_dpio_sideband_rw(INTEL_SIDEBAND_REG_WRITE,
+					IOSF_PORT_GPIO_NC, reg, &val);
+}
+
+static inline u32 vlv_gps_core_read(u32 reg)
+{
+	u32 val;
+	intel_adf_pci_sideband_rw(INTEL_SIDEBAND_REG_READ,
+					IOSF_PORT_GPS_CORE, reg, &val);
+	return val;
+}
+
+static inline void vlv_gps_core_write(u32 reg, u32 val)
+{
+	intel_adf_pci_sideband_rw(INTEL_SIDEBAND_REG_WRITE,
+					IOSF_PORT_GPS_CORE, reg, &val);
+}
+
+static inline u32 vlv_cck_read(u32 reg)
+{
+	u32 val;
+	intel_adf_pci_sideband_rw(INTEL_SIDEBAND_REG_READ, IOSF_PORT_CCK,
+			       reg, &val);
+	return val;
+}
+
+static inline void vlv_cck_write(u32 reg, u32 val)
+{
+	intel_adf_pci_sideband_rw(INTEL_SIDEBAND_REG_WRITE, IOSF_PORT_CCK,
+			       reg, &val);
+}
+
 bool vlv_intf_screen_connected(struct intel_pipe *pipe);
 u32 vlv_intf_vsync_counter(struct intel_pipe *pipe, u32 interval);
 extern int pipe_mode_set(struct intel_pipe *pipe,
-- 
1.7.9.5

